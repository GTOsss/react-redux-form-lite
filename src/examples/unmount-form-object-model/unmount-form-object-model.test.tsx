import React from 'react';
import {mount} from 'enzyme';
import {Provider} from 'react-redux';
import {createStore, applyMiddleware, combineReducers} from 'redux';
import ReduxThunkTester from 'redux-thunk-tester';
import UnmountForm from './unmount-form-object-model';
import {reducer} from '../../index';

const renderComponent = () => {
  const reduxThunkTester = new ReduxThunkTester();

  const store = createStore(
    combineReducers({reduxForm: reducer}),
    applyMiddleware(reduxThunkTester.createReduxThunkHistoryMiddleware()),
  );

  const component = mount(
    <Provider store={store}><UnmountForm /></Provider>,
  );

  return {reduxThunkTester, store, component};
};

describe('<UnmountFormObjectModel />', () => {
  test('Render UnmountForm (inputs)', () => {
    const {component} = renderComponent();
    expect(component.find('input')).toMatchSnapshot();
  });

  test('Render UnmountForm after unmount form', () => {
    const {component} = renderComponent();

    component.find('button').at(1).simulate('click');
    expect(component).toMatchSnapshot();
  });

  test('Render UnmountForm (store)', () => {
    const {store} = renderComponent();

    expect(store.getState().reduxForm.example).toMatchSnapshot();
  });

  test('Unmount form (store)', () => {
    const {component, store} = renderComponent();

    component.find('button').at(1).simulate('click');
    expect(store.getState().reduxForm.example).toMatchSnapshot();
  });

  test('Unmount form and mount form (store)', () => {
    const {component, store} = renderComponent();

    component.find('button').at(1).simulate('click');
    component.find('button').at(0).simulate('click');
    expect(store.getState().reduxForm.example).toMatchSnapshot();
  });
});
