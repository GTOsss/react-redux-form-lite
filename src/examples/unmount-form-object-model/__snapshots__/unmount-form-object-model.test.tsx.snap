// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<UnmountFormObjectModel /> Render UnmountForm (inputs) 1`] = `
Array [
  <input
    name="profile.firstName"
    onBlur={[Function]}
    onChange={[Function]}
    onFocus={[Function]}
    placeholder="First Name"
    type="text"
    value=""
  />,
  <input
    name="profile.lastName"
    onBlur={[Function]}
    onChange={[Function]}
    onFocus={[Function]}
    placeholder="Last Name"
    type="text"
    value=""
  />,
]
`;

exports[`<UnmountFormObjectModel /> Render UnmountForm (store) 1`] = `
Object {
  "form": Object {
    "activeField": "",
    "blurred": false,
    "changed": false,
    "errorsMap": Object {},
    "focused": false,
    "hasErrors": false,
    "hasWarnings": false,
    "submitted": false,
    "warningsMap": Object {},
  },
  "meta": Object {
    "profile": Object {
      "firstName": Object {
        "active": false,
        "blurred": false,
        "changed": false,
        "error": "",
        "focused": false,
        "warning": "",
      },
      "lastName": Object {
        "active": false,
        "blurred": false,
        "changed": false,
        "error": "",
        "focused": false,
        "warning": "",
      },
    },
  },
  "values": Object {
    "profile": Object {
      "firstName": undefined,
      "lastName": undefined,
    },
  },
}
`;

exports[`<UnmountFormObjectModel /> Render UnmountForm after unmount form 1`] = `
<Provider
  store={
    Object {
      "dispatch": [Function],
      "getState": [Function],
      "replaceReducer": [Function],
      "subscribe": [Function],
      Symbol(observable): [Function],
    }
  }
>
  <UnmountForm>
    <div>
      <button
        onClick={[Function]}
        type="button"
      >
        Toggle show form
      </button>
    </div>
  </UnmountForm>
</Provider>
`;

exports[`<UnmountFormObjectModel /> Unmount form (store) 1`] = `Object {}`;

exports[`<UnmountFormObjectModel /> Unmount form and mount form (store) 1`] = `
Object {
  "form": Object {
    "activeField": "",
    "blurred": false,
    "changed": false,
    "errorsMap": Object {},
    "focused": false,
    "hasErrors": false,
    "hasWarnings": false,
    "submitted": false,
    "warningsMap": Object {},
  },
  "meta": Object {
    "profile": Object {
      "firstName": Object {
        "active": false,
        "blurred": false,
        "changed": false,
        "error": "",
        "focused": false,
        "warning": "",
      },
      "lastName": Object {
        "active": false,
        "blurred": false,
        "changed": false,
        "error": "",
        "focused": false,
        "warning": "",
      },
    },
  },
  "values": Object {
    "profile": Object {
      "firstName": undefined,
      "lastName": undefined,
    },
  },
}
`;
